firebase-firestore/ktx/src/test/kotlin/com/google/firebase/firestore/ktx/JavaLibKtxSupportTests.kt

```diff
@@ -0,0 +1,195 @@
+    }
+
+    @Test
+    fun `java converted Ktx DocumentId annotations should be seen during Ktx serialization`() {
+        @Serializable
+        data class AnnotationTest(@DocumentId val docId: String? = null)
+
+        val annotations = AnnotationTest.serializer().descriptor.getElementAnnotations(0)
+        assertThat(annotations).hasSize(1)
+        val annotation = annotations[0]
+        assertThat(annotation::class.java).isAssignableTo(DocumentId::class.java)
+    }
+
+    @Test
+    fun `java converted Ktx ThrowOnExtraProperties annotations should be seen during Ktx serialization`() {
+        @Serializable
+        @ThrowOnExtraProperties
+        data class AnnotationTest(val docId: String? = null)
+
+        val annotations = AnnotationTest.serializer().descriptor.annotations
+        assertThat(annotations).hasSize(1)
+        val annotation = annotations[0]
+        assertThat(annotation::class.java).isAssignableTo(ThrowOnExtraProperties::class.java)
+    }
+
+    @Test
+    fun `java converted Ktx IgnoreExtraProperties annotations should be seen during Ktx serialization`() {
+        @Serializable
+        @IgnoreExtraProperties
+        data class AnnotationTest(val docId: String? = null)
+
+        val annotations = AnnotationTest.serializer().descriptor.annotations
+        assertThat(annotations).hasSize(1)
+        val annotation = annotations[0]
+        assertThat(annotation::class.java).isAssignableTo(IgnoreExtraProperties::class.java)
+    }
+
+    @Test
+    fun `java GeoPoint Ktx's serializer should be seen during Ktx serialization`() {
+        @Serializable
+        data class JavaGeoPoint(val geoPoint: GeoPoint)
+
+        val geoPointObj = JavaGeoPoint.serializer().descriptor.getElementDescriptor(0)
+        assertThat(geoPointObj.serialName).isEqualTo("__GeoPointSerializer__")
+    }
+
+    @Test
+    fun `java Timestamp Ktx's serializer should be seen during Ktx serialization`() {
+        @Serializable
+        data class JavaTimestamp(val timestamp: Timestamp)
+
+        val timestampObj = JavaTimestamp.serializer().descriptor.getElementDescriptor(0)
+        assertThat(timestampObj.serialName).isEqualTo("__TimestampSerializer__")
+    }
+
+    @Test
+    fun `java DocumentReference Ktx's serializer should be seen during Ktx serialization`() {
+        @Serializable
+        data class JavaDocRef(val docRef: DocumentReference)
+
+        val docRefObj = JavaDocRef.serializer().descriptor.getElementDescriptor(0)
+        assertThat(docRefObj.serialName).isEqualTo("__DocumentReferenceSerializer__")
+    }
+
+    @Serializable
+    private data class User(val name: String)
+
+    @Serializable
+    private data class Project(val name: String, val owner: User, val votes: Int)
+
+    private val data = Project("kotlinx.serialization", User("kotlin"), 9000)
+
+    @Test
+    fun `ktx serialization unit test list encoder is working`() {
```

Comment:
Not entirely sure what you are testing here? You introduced a test implementation of the encoder/decoder, and you are testing that test implementation?

Reply:
Yes, Yes, this is exactly what I did. I thought I need to test these test implementation of encoder/decoder works on primitive types, Then, I can use these pair of encoder/decoder to assert that the `Serializer`s I obtained from Java are valid (the descriptors and methods are indeed able to encode/decode the data type).

It seems you think these pre-checks for primitives are not necessary, I will just delete the tests.